collada_parser/src/collada_parser.cpp

```diff
@@ -176,7 +176,11 @@ class ColladaModelReader : public daeErrorHandler
         USERDATA(double scale) : scale(scale) {
         }
         double scale;
+#if __cplusplus <= 199711L
```

Comment:
The decision for boost::shared_ptr vs. std::shared_ptr shouldn't implicitly depend on the c++ version, but it should be an deliberative decision of the developer. You should declare a type VoidSharedPtr for this.